lib/stages/basebandReadout.cpp

```diff
@@ -470,13 +527,32 @@ basebandDumpData basebandReadout::get_data(uint64_t event_id, int64_t trigger_st
         int64_t frame_ind_end =
             std::min(data_end_fpga - frame_fpga_seq, (int64_t)_samples_per_data_set);
         int64_t data_ind_start = frame_fpga_seq - data_start_fpga + frame_ind_start;
-        // The following copy has 0 length unless there is a missing frame.
         nt_memset(&dump.data[next_data_ind * _num_elements], 0,
                   (data_ind_start - next_data_ind) * _num_elements);
-        // Now copy in the frame data.
-        nt_memcpy(&dump.data[data_ind_start * _num_elements],
-                  &buf_data[frame_ind_start * _num_elements],
-                  (frame_ind_end - frame_ind_start) * _num_elements);
+        if (_num_local_freq == 1) {
+            using namespace std::chrono;
+            milliseconds ms_before =
+                duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+            nt_memcpy(&dump.data[data_ind_start * _num_elements],
+                      &buf_data[frame_ind_start * _num_elements],
+                      (frame_ind_end - frame_ind_start) * _num_elements);
+            milliseconds ms_after =
+                duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+
+        } else if (_num_local_freq > 1) {
+            using namespace std::chrono;
+            milliseconds ms_before =
+                duration_cast<milliseconds>(system_clock::now().time_since_epoch());
+            // INFO("Right before memcpy()");
+            for (int timeidx = 0; timeidx < (frame_ind_end - frame_ind_start); timeidx++) {
+                memcpy(&dump.data[(data_ind_start + timeidx) * _num_elements],
+                       &buf_data[((frame_ind_start + timeidx) * _num_local_freq + freqidx)
+                                 * _num_elements],
+                       _num_elements);
+            }
+            milliseconds ms_after =
+                duration_cast<milliseconds>(system_clock::now().time_since_epoch());
```

Comment:
This time-keeping is not really used, is it?

Reply:
Yes, that's right. I will remove it.